### ISSUE TITLE
Add GTF for allowing getting raw payload of the transaction

### ISSUE THREAD

**Issue by luizstacio** [FuelLabs/fuel-specs#554]
It should be possible to get the entire transaction bytecode at execution time. This allows developers to creating predicates or scripts that can create different transactions hashs, with specific inputs zeroed out.

To do;
- Implement GTF on the VM to return entire transaction bytecode
- Implement tooling on Sway to allow getting a transaction type with good access for manipulation and hashing.

**Comment by xgreenx** [comment]
What is "transaction bytecode" here? Do you mean serialized transaction in the memory? Or something else?

**Comment by luizstacio** [comment]
Yes the serialized transaction in the memory, The entire TX bytes from tx type to witnesses bytes.

**Comment by xgreenx** [comment]
Including witnesses? What is the purpose of this feature? Do you plan to take the hash of it?

**Comment by luizstacio** [comment]
> Including witnesses? What is the purpose of this feature? Do you plan to take the hash of it?

Yes the idea is to been able to create different hashing schemas of the transaction without some specific fields.

**Comment by xgreenx** [comment]
Based on the internal discussion: We need to add a getter for the pointer where transactions start in the memory and its size.
